pkg/metrics/handler.go

```diff
@@ -0,0 +1,16 @@
+package metrics
+
+import "net/http"
+
+func AddHeaderAsCookieMiddleware(next http.Handler) http.Handler {
```

Comment:
The name here is a bit misleading since its a handler

Reply:
We should probably decouple it into the handler and the actual function that does set the header

Reply:
But it's a handler that gets another handler as a parameter and calls it (`next.ServeHTTP(w, r)`) after it changes the request.

It's a term I would def. use in JS. Would you prefer `AddHeaderAsCookieHandler` anyway?